doc/tools.md

```diff
@@ -1,17 +1,57 @@
 Tools
 =====
-
-Content
--------
-  1. [dlite-getuuid](#dlite-getuuid)
-  2. [dlite-codegen](#dlite-codegen)
+DLite comes with a small set of tools.
 
 
 dlite-getuuid
 -------------
-xxx
+This is a handy small tool for generating a random UUID or getting the UUID corresonding to an URI.
+
+Example usage:
+
+```console
+# No argument, generate a random UUID
+$ dlite-getuuid
+b7ffeb70-5ffe-43de-91e0-941244d21d68
+
+# Calculate the UUID corresponding to an URI
+$ dlite-getuuid http://onto-ns.com/meta/example/0.1/Example
+022b27f0-dd91-516d-83cb-94db8f65e42b
+
+# It recognises if the argument is already a valid UUID
+$ dlite-getuuid b7ffeb70-5ffe-43de-91e0-941244d21d68
+b7ffeb70-5ffe-43de-91e0-941244d21d68
+
+# ...even when the the UUID is encoded in the form `<META_URI>/<UUID>` optionally followed by a slash (/) or hash (#)
+$ dlite-getuuid http://onto-ns.com/meta/0.1/Collection/b7ffeb70-5ffe-43de-91e0-941244d21d68#
+b7ffeb70-5ffe-43de-91e0-941244d21d68
+```
 
 
 dlite-codegen
 -------------
-xxx
+This is a template-based code generator for C and Fortran (it is not needed for Python due to its dynamic nature).
```

Comment:
Based on the descriptions of the pre-defined templates, it sounds like dlite-codegen is essential for working with DLite in C and Fortran (?). If so, please consider pointing out this importance already in its introduction.